packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart

```diff
@@ -0,0 +1,118 @@
+      as IntegrationTestWidgetsFlutterBinding;
+  // resolves issue on iOS. See: https://github.com/flutter/flutter/issues/89651
+  binding.deferFirstFrame();
+
+  final authenticator = MaterialApp(
+    home: Authenticator(child: Container()),
+  );
+
+  String phone = generatePhone();
+  group('sign-up-with-phone', () {
+    // Given I'm running the example "ui/components/authenticator/sign-up-with-username"
+    setUpAll(() async {
+      await loadConfiguration(
+        'ui/components/authenticator/sign-up-with-phone',
+        additionalConfigs: [AmplifyAPI()],
+      );
+    });
+
+    // Scenario: Login mechanism set to "email"
+    testWidgets('Login mechanism set to "phone"', (tester) async {
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+      signUpPage.expectUserNameIsPresent(usernameLabel: 'Phone Number');
+    });
+
+    // Scenario: "Email" is included from `aws_cognito_verification_mechanisms`
+    testWidgets('"Email" is included from aws_cognito_verification_mechanisms',
+        (tester) async {
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+      signUpPage.expectEmailIsPresent();
+    });
+
+    // Scenario: Sign up with valid phone number & password
+    testWidgets('Sign up a new phone number & password', (tester) async {
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      ConfirmSignUpPage confirmSignUpPage = ConfirmSignUpPage(tester: tester);
+
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+
+      //   // TODO: Clarify requirements
+      //   // Given I intercept '{ "headers": { "X-Amz-Target": "AWSCognitoIdentityProviderService.SignUp" } }'
+      //   // with fixture "sign-up-with-email"
+
+      final password = generatePassword();
+      final email = generateEmail();
+
+      // When I select my country code with status "+1" and phone with '55501xx'
+      await signUpPage.enterUsername(phone);
+
+      // And I type my password
+      await signUpPage.enterPassword(password);
+
+      // And I confirm my password
+      await signUpPage.enterPasswordConfirmation(password);
+
+      // And I type my "email" with a randomized email
+      await signUpPage.enterEmail(email);
+
+      // And I click the "Create Account" button
+      await signUpPage.submitSignUp();
+
+      // Then I see "Confirmation Code"
+      await confirmSignUpPage.expectConfirmSignUpIsPresent();
+      confirmSignUpPage.expectConfirmationCodeIsPresent();
+      await deleteUser(phone);
```

Comment:
I would put this in `tearDownAll`

Reply:
The to-be-deleted user is only created on this final test.

Reply:
Ah, my mistake. Sounds good